net/message/message.go

```diff
@@ -104,12 +104,12 @@ func (m *QueryResponse) Type() byte {
 // SendKeysMessage is sent by both parties to each other to initiate the protocol
 type SendKeysMessage struct {
 	OfferID            types.Hash              `json:"offerID"`
-	ProvidedAmount     *apd.Decimal            `json:"providedAmount"`
-	PublicSpendKey     *mcrypto.PublicKey      `json:"publicSpendKey"`
+	ProvidedAmount     *apd.Decimal            `json:"providedAmount" validate:"required"`
+	PublicSpendKey     *mcrypto.PublicKey      `json:"publicSpendKey" validate:"required"`
 	PublicViewKey      *mcrypto.PublicKey      `json:"publicViewKey"`
 	PrivateViewKey     *mcrypto.PrivateViewKey `json:"privateViewKey"`
```

Comment:
@noot After your current work is complete, which fields will be optional by the maker and/or taker, but required by the other party in this structure?  If they don't line up, should we consider using 2 different types for better self-documenting code?  We already changed the top level functions to use an interface, so the impact might not be that bad.

Reply:
`PublicViewKey` will be gone, only `PrivateViewKey` will be here, so the messages will actually be the same for both parties which is nicer